File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.split(": ")[1].trim();
    }

    public static String logLevel(String logLine) {
      	String level = logLine.split(": ")[0];
        return level.substring(1,level.length()-1).toLowerCase();    
    }

    public static String reformat(String logLine) {
        String level = LogLevels.logLevel(logLine);
        String message = LogLevels.message(logLine);
		String reformattedString = message + " (" + level + ")";
        return reformattedString;    
    }
}

File: cpp/log-levels/log_levels.cpp
#include <string>
using namespace std;
namespace log_line {
    string message(string logMessage){
        string deliminator {" "};
        int index = logMessage.find(deliminator);
        string message = logMessage.substr(index + 1);
        return message;
    }

    string log_level(string logMessage) {
        string deliminator {"]: "};
        int index = logMessage.find(deliminator);
        string level = logMessage.substr(1, index-1);
        return level;
    }

    string reformat(string logMessage) {
        return message(logMessage) + " (" + log_level(logMessage) + ")";
    }
} // namespace log_line